server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -69,7 +69,10 @@
     private final Map<String, CircuitBreaker> breakers;
 
     public static final Setting<Boolean> USE_REAL_MEMORY_USAGE_SETTING =
-        Setting.boolSetting("indices.breaker.total.use_real_memory", true, Property.NodeScope);
+        Setting.boolSetting("indices.breaker.total.use_real_memory", settings -> {
+            // turn real memory circuit breaker off for ParallelGC
+            return String.valueOf(Booleans.parseBoolean(JvmInfo.jvmInfo().useParallelGC()) == false);
```

Comment:
In #46751, a check is made purely against the parent breaker, relying on this being the real memory circuit breaker. This PR disables that PR by default for small heaps.

I seem to remember seeing that approach (just checking the parent breaker, now that we have real memory circuit breaker) has come up a few times.